studio/studio.py

```diff
@@ -237,6 +245,171 @@ async def export(q: Q):
     os.remove("app.zip")
 
 
+def get_package_dialog_items():
+    file = open('project/requirements.txt', 'r') if os.path.exists('project/requirements.txt') else None
+    packages_installed = os.stat("project/requirements.txt").st_size > 0 if file else False
+    return [
+        ui.text_l('Installed packages'),
+        ui.inline(
+            direction='column', align='start',
+            # List of installed packages.
+            items=[
+                ui.inline(align='end', items=[
+                    ui.textbox(name='package_name', value=package_name, readonly=True),
+                    ui.textbox(name='package_version', value=package_version, readonly=True),
+                    ui.button('remove_package', label='Remove', value=package_name),
+                ]) for package_name, package_version in
+                [package.removesuffix('\n').split('==') for package in file.readlines()]
+            ],
+        ) if packages_installed else ui.text(name='no_packages', content='No packages installed.'),
+        ui.inline(
+            align='end',
+            justify='center',
+            height='60px',
+            items=[ui.buttons(items=[
+                ui.button(name='show_add_package_fields', label='Add package', icon='Add', primary=True, width='200px'),
+                ui.button(name='show_add_requirements', label='Add from requirements', icon='AddToShoppingList',
+                          width='200px')
+            ], justify='center')]
+        ),
+    ]
+
+
+def get_output(output: str) -> str:
+    return f'''
+```
+{output}
+```
+'''
+
+
+async def show_progress(q: Q, msg: str):
+    q.page['meta'].dialog.items = [
+        ui.progress(label=msg),
+        ui.expander(name='console_expander', label='Detail', items=[
+            ui.text(name='console_out', content=get_output(''), width='100%'),
+        ]),
```

Comment:
The UX can be improved here. After hitting the install button, the (pip) command output should be streamed automatically instead of showing a loading indicator and then require extra click.